### PR TITLE
roachtest: update pg_regress diff

### LINKED ISSUE #119626: roachtest: pg_regress failed
roachtest.pg_regress [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14148216?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14148216?buildTab=artifacts#/pg_regress) on master @ [c9c3cc5f3c3a4a6ab556f4b9d5b6ec0381901bdb](https://github.com/cockroachdb/cockroach/commits/c9c3cc5f3c3a4a6ab556f4b9d5b6ec0381901bdb):

```
(pg_regress.go:451).runPGRegress: Regression diffs do not match expected output:
--- testdata/regression.diffs
+++ regression.diffs
@@ -226107,7 +226107,7 @@
 +github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:361: func2()
 +github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:364: runPlanInsidePlan()
 +github.com/cockroachdb/cockroach/pkg/sql/routine.go:276: func2()
-+github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/scalar.go:1230: func2()
++github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/scalar.go:1228: func2()
 +github.com/cockroachdb/cockroach/pkg/sql/routine.go:239: startInternal()
 +github.com/cockroachdb/cockroach/pkg/sql/routine.go:180: Start()
 +github.com/cockroachdb/cockroach/pkg/sql/rowexec/project_set.go:208: nextInputRow()
@@ -226160,7 +226160,7 @@
 +github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:361: func2()
 +github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:364: runPlanInsidePlan()
 +github.com/cockroachdb/cockroach/pkg/sql/routine.go:276: func2()
-+github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/scalar.go:1230: func2()
++github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/scalar.go:1228: func2()
 +github.com/cockroachdb/cockroach/pkg/sql/routine.go:239: startInternal()
 +github.com/cockroachdb/cockroach/pkg/sql/routine.go:180: Start()
 +github.com/cockroachdb/cockroach/pkg/sql/rowexec/project_set.go:208: nextInputRow()
If the diff is expected, copy regression.diffs from the test artifacts to pkg/cmd/roachtest/testdata/regression.diffs
test artifacts and logs in: /artifacts/pg_regress/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14148216/pg-regress/1708885676635/1708886449953)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pg_regress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-36246

### PR DESCRIPTION
**Pull request by yuzefovich**
This commit updates the diff based on recent line number changes. The good news is that the internal errors that print the stack trace should be fixed shortly (PR is already out).

I also spent some time trying to understand what would it take to make `pg_regress` a part of the local roachtest run in CI (so that we don't get these nightly failures and, instead, update the diff as part of the change that triggers the diff), but it seems to be a no-go for two reasons:
- we generally want to make CI run faster, and pg_regress takes on the order of 10 minutes on a gceworker which would make it the slowest roachtest that is part of CI I think
- we run local roachtests in CI inside of the docker container, which at least currently doesn't have `sudo` command to install things like postgres needed for this roachtest.

So I think for now we'll need to either update the diff based on the nightly failure or proactively manually kick of a run of pg_regress when we know that the diff might need to be changed.

Fixes: #119626.

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/testdata/regression.diffs [hunk 1]
```diff
@@ -162375,10 +162376,10 @@ diff -U3 --label=/mnt/data1/postgres/src/test/regress/expected/collate.windows.w
 +    b text COLLATE "C"
 +)
 +^
-+INSERT INTO collate_test1 VALUES (1, 'abc'), (2, '�bc'), (3, 'bbc'), (4, 'ABC');
++INSERT INTO collate_test1 VALUES (1, 'abc'), (2, '�bc'), (3, 'bbc'), (4, 'ABC');
```
Comment (rytaft) [inline pkg/cmd/roachtest/testdata/regression.diffs:162379]: what changed to cause this diff?

**Review by rytaft — APPROVED** [review]
LGTM, just one question below

File: pkg/cmd/roachtest/testdata/regression.diffs (hunk 1 above)
Comment (yuzefovich) [reply pkg/cmd/roachtest/testdata/regression.diffs]: I'm not quite sure - perhaps the last time we simply downloaded artifacts from the failed nightly run and this time I manually kicked of a run. The difference seems to be with unicode characters. I reverted these changes.

**Review by yuzefovich — COMMENTED** [review]
TFTR!

bors r+

**Pull request merged**